misc/python/materialize/mzcompose.py

```diff
@@ -66,291 +64,139 @@
-            say(f"WARNING: services list specified with -w, ignoring: {serv}")
-        comp.run_workflow(workflow)
-    else:
-        comp.up(list(services))
-
-
-@cli.command()
-@click.option("-w", "--workflow", help="The name of a workflow to run")
-@click.argument("composition")
-@click.argument("services", nargs=-1)
-def run(composition: str, workflow: Optional[str], services: Iterable[str],) -> None:
-    """Conduct a docker-composed set of services
-
-    With the --workflow flag, perform all the steps in the workflow together.
-    """
-    ui.warn_docker_resource_limits()
-    comp = Composition.find(composition)
-    if workflow is not None:
-        say(f"Executing {comp.name} -> {workflow}")
-        if services:
-            serv = " ".join(services)
-            say(f"WARNING: services list specified with -w, ignoring: {serv}")
-        comp.run_workflow(workflow)
-    else:
-        comp.run(list(services))
-
-
-@cli.command()
-@click.argument("composition")
-def build(composition: str) -> None:
-    Composition.find(composition).build()
-
-
-@cli.command()
-@click.option("-v", "--volumes", is_flag=True, help="Also destroy volumes")
-@click.argument("composition")
-def down(composition: str, volumes: bool) -> None:
-    comp = Composition.find(composition)
-    comp.down(volumes)
-
-
-@cli.command()
-@click.argument("composition")
-def ps(composition: str) -> None:
-    comp = Composition.find(composition)
-    comp.ps()
-
-
-# Non-mzcompose commands
-
-
-@cli.command()
-@click.argument("composition")
-def help_workflows(composition: str) -> None:
-    """Help on available workflows in DEMO"""
-    comp = Composition.find(composition)
-    print("Workflows available:")
-    for workflow in comp.workflows():
-        print(f"    {workflow.name}")
-
-
-@cli.command()
-@click.argument("composition")
-def nuke(composition: str) -> None:
-    """Destroy everything docker, stopping composition before trying"""
-    comp = Composition.find(composition)
-    comp.down()
-    cmds = ["docker system prune -af".split(), "docker volume prune -f".split()]
-    for cmd in cmds:
-        spawn.runv(cmd, capture_output=True)
-
-
-@cli.command()
-@click.argument("composition")
-@click.argument("service")
-def web(composition: str, service: str) -> None:
-    """
-    Attempt to open a service in a web browser
-
-    This parses the output of `mzconduct ps` and tries to find the right way to open a
-    web browser for you.
-    """
-    comp = Composition.find(composition)
-    comp.web(service)
-
-
-@cli.group()
-def show() -> None:
-    """Show properties of a composition"""
-
-
-@show.command()
-@click.argument("composition")
-def dir(composition: str) -> None:
-    """Show the directory that this composition is in"""
-    print(str(Composition.find(composition).path))
-
-
-# Composition Discovery
-
-
 class Composition:
-    """Information about an mzcompose instance
+    """A set of parsed mzcompose.yml files."""
```

Comment:
This is the set of mzcompose.yml files for a single directory, right?

Reply:
Good point, yeah.